README.md

```diff
@@ -1787,86 +1777,57 @@ message has the following structure:
 
 ```protobuf
 message AddonAvailable {
-    DownloadableFileList files = 1;
+    DownloadableFile file = 1;
     bytes hash = 2;
 }
 ```
 
 
 TODO: do we need other fields, e.g. addon version or description?
 
-<h4 id="files">files</h4>
-
-
-The list of files in the addon. The map of addon files. Keys are file names.
-
-<h4 id="hash">hash</h4>
-
-
-The aggregate hash of the addon. SHOULD be calculated based on file names and
-content of all files in the addon.
-
-<h3 id="downloadablefilelist-message">DownloadableFileList Message</h3>
-
-
-A list of files that can be downloaded. The list's aggregate hash and the hash
-of each individual file is provided so that the downloading can be skipped if
-the Agent already has the collection or the individual files. The message has
-the following structure:
-
-
-```protobuf
-message DownloadableFileList {
-    map<string,DownloadableFile> files = 1;
-    bytes hash = 2;
-}
-```
-
-
-<h4 id="files">files</h4>
+<h4 id="file">file</h4>
 
-
-The map of addon files. Keys are file names.
+The downloadable file of the addon.
 
 <h4 id="hash">hash</h4>
 
-
-The aggregate hash of the addon. SHOULD be calculated based on file names and
-content of all files in the addon.
+The hash of the addon. SHOULD be calculated based on addon name, file name and
+content of the file of the addon.
 
 <h3 id="downloadablefile-message">DownloadableFile Message</h3>
 
-
 The message has the following structure:
 
-
 ```protobuf
 message DownloadableFile {
-    string download_url = 1;
-    bytes hash = 2;
+    string file_name = 1;
```

Comment:
Do we need the file name? What if we remove it? For addons we have the addon name to use the name of the local file to download to. For agent packages the agent is free to use a hard-coded name. It is not clear what we gain by having the file name in the message.

Reply:
I agree that `file_name` seems to be providing little value. If we need the original file name for whatever purpose then we should be also able to extract it from the  `download_url` or [Content-Disposition](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) header

Reply:
Removed.